src/components/RightSidebar/TableOfContents.tsx

```diff
@@ -19,28 +19,61 @@ const TableOfContents: FunctionalComponent<Props> = ({ headers = [], labels }) =
 		if (!toc.current) return;
 
 		const setCurrent: IntersectionObserverCallback = (entries) => {
+			const intersectingEntries = [];
+
 			for (const entry of entries) {
-				if (entry.isIntersecting) {
-					setCurrentID(entry.target.id);
-					break;
-				}
+				if (entry.isIntersecting) intersectingEntries.push(entry);
+			}
+
+			if (intersectingEntries.length === 1) {
+				setCurrentID(intersectingEntries[0].target.id);
+				scrollIntoView();
+			} else if (intersectingEntries.length > 1) {
+				const sortedEntries = intersectingEntries.sort((a, b) => {
+					return getHeadingIndex(a.target.id) - getHeadingIndex(b.target.id);
+				});
+				setCurrentID(sortedEntries[0].target.id);
+				scrollIntoView();
 			}
 		};
 
 		const observerOptions: IntersectionObserverInit = {
 			// Negative top margin accounts for `scroll-margin`.
 			// Negative bottom margin means heading needs to be towards top of viewport to trigger intersection.
-			rootMargin: '-100px 0% -66%',
+			rootMargin: '-100px 0px -40% 0px',
```

Comment:
```suggestion
			rootMargin: '-100px 0px -40% 0px',
```
After a bit of playing, it appears a wee bit off, I'd need to spin this up and test it properly,